### ISSUE TITLE
compass -w fails with 1.9.1-p378

### ISSUE THREAD

**Issue by norman** [Compass/compass#97]
Using the "watch" feature works with 1.8.7 (EE), but fails with Ruby 1.9.1-p378:

My compass.config file:

```
project_path    = File.dirname(__FILE__)
sass_dir        = "views"
css_dir         = "public/stylesheets"
images_dir      = "public/images"
javascripts_dir = "public/js"
output_style    = "compact"
```

Output of running `compass`:

```
$ compass -c compass.config -w --trace
>>> Compass is watching for changes. Press Ctrl-C to Stop.
TypeError on line ["969"] of /Users/norman/.rvm/rubies/ruby-1.9.1-p378/lib/ruby/1.9.1/pathname.rb: can't convert Pathname into String
  /Users/norman/.rvm/rubies/ruby-1.9.1-p378/lib/ruby/1.9.1/pathname.rb:969:in `glob'
  /Users/norman/.rvm/gems/ruby-1.9.1-p378/gems/compass-0.8.17/lib/vendor/fssm/state.rb:49:in `add_glob'
  /Users/norman/.rvm/gems/ruby-1.9.1-p378/gems/compass-0.8.17/lib/vendor/fssm/state.rb:45:in `block in snapshot'
  /Users/norman/.rvm/gems/ruby-1.9.1-p378/gems/compass-0.8.17/lib/vendor/fssm/state.rb:45:in `each'
  /Users/norman/.rvm/gems/ruby-1.9.1-p378/gems/compass-0.8.17/lib/vendor/fssm/state.rb:45:in `snapshot'
  /Users/norman/.rvm/gems/ruby-1.9.1-p378/gems/compass-0.8.17/lib/vendor/fssm/state.rb:37:in `recache'
  /Users/norman/.rvm/gems/ruby-1.9.1-p378/gems/compass-0.8.17/lib/vendor/fssm/state.rb:9:in `refresh'
  /Users/norman/.rvm/gems/ruby-1.9.1-p378/gems/compass-0.8.17/lib/vendor/fssm/backends/polling.rb:10:in `add_path'
  /Users/norman/.rvm/gems/ruby-1.9.1-p378/gems/compass-0.8.17/lib/vendor/fssm/monitor.rb:18:in `path'
  /Users/norman/.rvm/gems/ruby-1.9.1-p378/gems/compass-0.8.17/lib/compass/commands/watch_project.rb:26:in `block (2 levels) in perform'
  /Users/norman/.rvm/gems/ruby-1.9.1-p378/gems/compass-0.8.17/lib/compass/commands/watch_project.rb:25:in `each'
  /Users/norman/.rvm/gems/ruby-1.9.1-p378/gems/compass-0.8.17/lib/compass/commands/watch_project.rb:25:in `block in perform'
  /Users/norman/.rvm/gems/ruby-1.9.1-p378/gems/compass-0.8.17/lib/vendor/fssm.rb:19:in `call'
  /Users/norman/.rvm/gems/ruby-1.9.1-p378/gems/compass-0.8.17/lib/vendor/fssm.rb:19:in `monitor'
  /Users/norman/.rvm/gems/ruby-1.9.1-p378/gems/compass-0.8.17/lib/compass/commands/watch_project.rb:24:in `perform'
  /Users/norman/.rvm/gems/ruby-1.9.1-p378/gems/compass-0.8.17/lib/compass/commands/base.rb:15:in `execute'
  /Users/norman/.rvm/gems/ruby-1.9.1-p378/gems/compass-0.8.17/lib/compass/commands/project_base.rb:19:in `execute'
  /Users/norman/.rvm/gems/ruby-1.9.1-p378/gems/compass-0.8.17/lib/compass/exec.rb:242:in `do_command'
  /Users/norman/.rvm/gems/ruby-1.9.1-p378/gems/compass-0.8.17/lib/compass/exec.rb:57:in `perform!'
  /Users/norman/.rvm/gems/ruby-1.9.1-p378/gems/compass-0.8.17/lib/compass/exec.rb:40:in `run!'
  /Users/norman/.rvm/gems/ruby-1.9.1-p378/gems/compass-0.8.17/bin/compass:8:in `<top (required)>'
  /Users/norman/.rvm/gems/ruby-1.9.1-p378/bin/compass:19:in `load'
  /Users/norman/.rvm/gems/ruby-1.9.1-p378/bin/compass:19:in `<main>'
```

Let me know if there is any more debugging you'd like me to do.

**Comment by bdtomlin** [comment]
Norman,
I had the same issue. See here for a solution: http://github.com/chriseppstein/compass/issues/issue/48

**Comment by chriseppstein** [comment]
This is fixed in v0.10, I suggest that you upgrade.

**Comment by bdtomlin** [comment]
Very cool. Thanks.
